netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,251 @@
+        return
+      }
+      val lastSeenEventId = prefs.getInt(lastSeenEventIdKey, -1); // Returns -1 if "event-id" is not found
+      println(s"lastSeenEventId: $lastSeenEventId, head of the list: ${jsonObjectList.head.eventId}")
+      if(jsonObjectList.head.eventId > lastSeenEventId){
+        println(s"Show this ${jsonObjectList.head.eventId}")
+      }
+      else {
+        println(s"Don't show this  ${jsonObjectList.head.eventId}")
+        return
+      }
+      val html = InfoFormatter.toInnerHtml(formattedString)
+
+      if (!jsonContent.trim.isEmpty) {
+        SwingUtilities.invokeLater(() => {
+          val editorPane: JEditorPane = new JEditorPane {
+            self =>
+            addFocusListener(new FocusListener {
+              def focusGained(fe: FocusEvent): Unit = {
+                FindDialog.watch(self)
+              }
+
+              def focusLost(fe: FocusEvent): Unit = {
+                if (!fe.isTemporary) FindDialog.dontWatch(self)
+              }
+            })
+            setDragEnabled(false)
+            setEditable(false)
+            setContentType("text/html")
+            setOpaque(false)
+            setText(html)
+            setCaretPosition(0)
+          }
+
+          val scrollPane = new JScrollPane(editorPane)
+          scrollPane.setPreferredSize(new Dimension(500, 400))
+          val panel = new JPanel(new BorderLayout())
+          panel.add(scrollPane, BorderLayout.CENTER)
+          val options: Array[AnyRef] = Array(I18N.gui.get("common.buttons.ok"))
+
+          val result = JOptionPane.showOptionDialog(
+            null, panel, I18N.gui.get("dialog.interface.newsNotificationTitle"),
+            JOptionPane.DEFAULT_OPTION, JOptionPane.PLAIN_MESSAGE, null, options, options(0)
+          )
+
+          // Check if OK was clicked (index 0 in options array) and hide the NotificationBanner
+          if (result == 0) {
+            setVisible(false) // Hide NotificationBanner
+            prefs.putInt("lastSeenEventId", jsonObjectList.head.eventId)
+          }
+        })
+      }
+    } catch {
+      case e: Exception => throw new Exception(s"Error in showJsonInDialog: ${e.getMessage}", e)
+    }
+  }
+
+
+  // Lazily initialize the Markdown parser and renderer (can be reused for multiple calls)
+  private lazy val markdownParser = Parser.builder().build()
+  private lazy val htmlRenderer = HtmlRenderer.builder().build()
+
+  def formatJsonObjectList(jsonObjectList: List[JsonObject]): String = {
+    jsonObjectList.map { obj =>
+      // Convert fullText from Markdown to HTML
+      val fullTextHtml = htmlRenderer.render(markdownParser.parse(obj.fullText))
+
+      // Format title, date, and the converted HTML fullText
+      s"<h3>* ${I18N.gui.get("dialog.interface.update")}: ${obj.date} -- ${obj.title}</h3>" +
+        s"$fullTextHtml"
+    }.mkString("<html>", "", "</html>")
+  }
+
+    // Static method to get the title of the head of jsonObjectList, with lazy initialization
```

Comment:
This method is not static, so probably not a good idea to say that in the comment or people will think it was supposed to be a static method.

Reply:
Done